photon-ml/src/main/scala/com/linkedin/photon/ml/function/IndividualObjectiveFunction.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.function
+
+import breeze.linalg.Vector
+
+import com.linkedin.photon.ml.data.LabeledPoint
+
+/**
+ * The base objective function used by IndividualOptimizationProblems. This function works with data locally as part of
+ * a single task (on a single executor).
+ */
+abstract class IndividualObjectiveFunction extends ObjectiveFunction with Serializable {
```

Comment:
I thought of `LocalObjectiveFunction` but Josh pointed out that the overall problem is still distributed, it's just only used on one machine.

I like `SingleObjectiveFunction` or `SingleNodeObjectiveFunction`. Please continue to voice your preferences, otherwise I'll just pick one of those two.